cis194/week4/zehua/src/Exercises.hs

```diff
@@ -0,0 +1,27 @@
+module Exercises where
+
+import Data.Bool (bool)
+
+-- ex1
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
+  | even x = (x - 2) * fun1 xs
+  | otherwise = fun1 xs
+
+fun1' :: [Integer] -> Integer
+fun1' = product . fmap (subtract 2) . filter even
```

Comment:
Just FYI here, because `-` carries 2 meanings. we need to write `((-) 2)` here for it to express subtracting.